### PR TITLE
Support for ILI9225 controller

### PR DESCRIPTION
**Pull request by michallenc**
## Summary
This PR adds basic support for ILI9225 controller and its support to 176x220 TFT display on Nucleo-F446RE board. 

## Testing
Functionality tested on Nucleo-F446RE board with framebuffer app example.

### CODE REVIEW

File: drivers/lcd/ili9225.c [hunk 1]
```diff
@@ -0,0 +1,744 @@
...
+ * Name: ili9225_setarea
+ *
+ * Description:
+ *   Set the rectangular area for an upcoming read or write from RAM.
+ *
+ ****************************************************************************/
+
+static void ili9225_setarea(FAR struct ili9225_dev_s *dev,
+                           uint16_t x0, uint16_t y0,
+                           uint16_t x1, uint16_t y1)
```
Comment (acassis) [inline drivers/lcd/ili9225.c:449]: uint16_t should under right below FAR

File: drivers/lcd/ili9225.c [hunk 2]
```diff
@@ -0,0 +1,744 @@
...
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: ili9225_select
+ *
+ * Description:
+ *   Select the SPI, locking and  re-configuring if necessary
```
Comment (acassis) [inline drivers/lcd/ili9225.c:222]: Please remove extra space: "and re-configuring"

File: drivers/lcd/ili9225.c [hunk 3]
```diff
@@ -0,0 +1,744 @@
...
+/****************************************************************************
+ * Name: ili9225_wrram
+ *
+ * Description:
+ *   Write to the driver's RAM.
+ *
+ ****************************************************************************/
+
+static void ili9225_wrram(FAR struct ili9225_dev_s *dev,
+                         FAR const uint16_t *buff, size_t size)
```
Comment (acassis) [inline drivers/lcd/ili9225.c:491]: Please put FAR aligned under FAR

File: drivers/lcd/ili9225.c [hunk 4]
```diff
@@ -0,0 +1,744 @@
...
+ * Name: ili9225_rdram
+ *
+ * Description:
+ *   Read from the driver's RAM.
+ *
+ ****************************************************************************/
+
+#ifndef CONFIG_LCD_NOGETRUN
+static void ili9225_rdram(FAR struct ili9225_dev_s *dev,
+                         FAR uint16_t *buff, size_t size)
```
Comment (acassis) [inline drivers/lcd/ili9225.c:510]: Ditto

File: drivers/lcd/ili9225.c [hunk 5]
```diff
@@ -0,0 +1,744 @@
...
+ *   row     - Starting row to write to (range: 0 <= row < yres)
+ *   col     - Starting column to write to (range: 0 <= col <= xres-npixels)
+ *   buffer  - The buffer containing the run to be written to the LCD
+ *   npixels - The number of pixels to write to the LCD
+ *             (range: 0 < npixels <= xres-col)
+ *
+ ****************************************************************************/
+
+static int ili9225_putrun(fb_coord_t row, fb_coord_t col,
+                         FAR const uint8_t *buffer, size_t npixels)
```
Comment (acassis) [inline drivers/lcd/ili9225.c:558]: ditto

**Comment by acassis** [comment]
Very well done @michallenc ! Kudos!!!

**Comment by michallenc** [comment]
Hi @acassis. Done, thanks for a quick review!

**Pull request merged**